### ISSUE TITLE
Use "pwsh.exe" on Windows instead of the old "powershell.exe" in a non-breaking way

### ISSUE THREAD

**Issue by frederikhors** [casey/just#1185]
I'm new to just. It's great! Really!

I'm a Windows 10 user and I'm using the "new" PowerShell 7 which has the executable name `pwsh.exe`.

I understood from the docs that I can set the below config on Windows to use Powershell instead of `sh.exe` (which is what I want since `sh.exe` on Windows is buggy for some things).

```justfile
set windows-powershell
```

What is missing today (for what I can read from docs at least) is a way to indicate to `Just` what Powershell version to use based on what executable to use:

1. the old buggy `powershell.exe` (which is currently used)
1. or the new `pwsh.exe` which you can download & install from Microsoft website (and maybe is already installed on Windows 11).

I tried with these lines too but obviously they aren't working and they have no sense at all since on my linux I would use the default shell for linux:

```justfile
shell := if os_family() == "windows" { "pwsh.exe" } else { "" }
set shell := ["{{shell}}", "-c"]
```

So I'm suggesting a non-breaking way to handle this using something like:

```justfile
set windows-pwsh
```

Do you agree?

**Comment by casey** [comment]
> I'm new to just. It's great! Really!

Thank you!

> the old buggy powershell.exe (which is currently used)

So I have the background, what's buggy about the binary named `powershell.exe`? Also, what version is it? Is `pwsh.exe` a different version?

> set windows-pwsh

That sounds reasonable to me, although it's unfortunate that there would be two very similar options.

**Comment by frederikhors** [comment]
Sorry, I used a wrong term here ("buggy").

`powershell.exe` is the PowerShell 6.

`pwsh.exe` is the NEW PowerShell 7, faster and better (I can use many bash-similar commands, I can even use the `&&` between commands).

> although it's unfortunate that there would be two very similar options

Maybe we can use the same but if `pwsh.exe` is found we can use it.

Although maybe this can be slower?

**Comment by casey** [comment]
Okay, I see. I think switching between `pwsh.exe` and `powershell.exe` automatically is probably a bad idea. As gross as it is, `set windows-pwsh` is probably the way to go.

**Comment by casey** [comment]
@frederikhors Just published 1.2.0, which has `set windows-shell`, which can be used to set the windows shell to whatever, including `pwsh.exe`. Let me know if it works for you!
